inc/class-wpseo-utils.php

```diff
@@ -483,35 +492,266 @@ public static function clear_transient_cache( $option ) {
 	 *
 	 * @param array $types Set of sitemap types to invalidate cache for.
 	 */
-	public static function clear_sitemap_cache( $types = array() ) {
-		global $wpdb;
+	public static function clear_sitemap_cache( array $types = array() ) {
+		// Filter out optional empty items.
+		$types = array_filter( $types );
+
+		// Clear all cache.
+		if ( empty( $types ) ) {
+			self::invalidate_sitemap_cache();
 
-		if ( wp_using_ext_object_cache() ) {
 			return;
 		}
 
-		if ( ! apply_filters( 'wpseo_enable_xml_sitemap_transient_caching', true ) ) {
-			return;
+		// Make sure the index cache always gets invalidated.
+		if ( ! in_array( '1', $types ) ) {
```

Comment:
Doest the `'1'` stand for something? Maybe it worth consider declaring it as a class constant.